ipv8/attestation/trustchain/community.py

```diff
@@ -386,29 +388,32 @@ def process_half_block(self, blk, peer):
 
         self.logger.info("Received request block addressed to us (%s)", blk)
 
-        # determine if we want to sign this block
-        if not self.should_sign(blk):
-            self.logger.info("Not signing block %s", blk)
-            return succeed(None)
+        def on_should_sign_outcome(should_sign):
+            if not should_sign:
+                self.logger.info("Not signing block %s", blk)
+                return None
```

Comment:
`addCallback` and `self.sign_block` return a `Deferred`. How are the unit tests even passing?